tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -1956,8 +1964,10 @@ class Subgraph {
       TF_LITE_ENSURE(logging_context, webnn_operands.at(filter_tensor_id).as<bool>());
       if (fc_params->keep_num_dims || input_tensor.dims->size != 2) {
         // Reshape input to 2D tensor
-        const int32_t n_inputs = input_channels;
-        std::vector<int32_t> new_input_shape = {-1, n_inputs};
+        const uint32_t n_inputs = input_channels;
+        const uint32_t size_of_input_shape = SizeOfShape(input_tensor.dims);
+        std::vector<uint32_t> new_input_shape =
+            {size_of_input_shape / n_inputs, n_inputs};
```

Comment:
Why not set `new_input_shape` to `[null, n_inputs]`?

Reply:
I just want to avoid including additional lib for absl::nullopt, which comes from "@com_google_absl//absl/".

Reply:
Should you use [`val::null()`](https://emscripten.org/docs/api_reference/val.h.html#_CPPv4N10emscripten10emscripten3val4nullEv)?

Reply:
Oh, thanks for reminder! I should use `val::null()`. I've submitted a new commit, please take another look, thanks!